tools/releaser/main.go

```diff
@@ -0,0 +1,289 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"os/signal"
+	"regexp"
+	"strings"
+)
+
+const (
+	_tmpBzl = "tmp.bzl"
+)
+
+func main() {
+	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt)
+	defer cancel()
+	if err := cli(ctx, os.Stderr, os.Args[1:]); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+func cli(ctx context.Context, stderr *os.File, args []string) error {
+	if len(args) > 1 || len(args) == 1 && strings.HasPrefix(strings.TrimLeft(args[0], "-"), "h") {
+		fmt.Println(`usage: go run tools/releaser/main.go
+
+This utility is intended to handle many of the steps to release a new version.
+
+Only run it from the root of the bazel-gazelle repository.`)
+		return nil
+	}
+
+	var verbose bool
+	if len(args) == 1 && strings.HasPrefix(strings.TrimLeft(args[0], "-"), "v") {
+		verbose = true
+	}
+
+	if verbose {
+		fmt.Println("Running initial go update commands")
+	}
+	initialCommands := []struct {
+		cmd  string
+		args []string
+	}{
+		{cmd: "go", args: []string{"get", "-t", "-u", "./..."}},
+		{cmd: "go", args: []string{"mod", "tidy", "-compat=1.16"}},
+		{cmd: "go", args: []string{"mod", "vendor"}},
+		{cmd: "find", args: []string{"vendor", "-name", "BUILD.bazel", "-delete"}},
+	}
+	for _, c := range initialCommands {
+		if out, err := exec.CommandContext(ctx, c.cmd, c.args...).CombinedOutput(); err != nil {
+			fmt.Println(string(out))
+			return err
+		}
+	}
+
+	workspace, err := os.ReadFile("WORKSPACE")
+	if err != nil {
+		return err
+	}
+
+	workspaceBuffer := bytes.NewBuffer(workspace)
+	workspaceWithoutDirectives := new(bytes.Buffer)
+
+	if verbose {
+		fmt.Println("Preparing temporary WORKSPACE without gazelle directives.")
+	}
+	directiveStart, err := getWorkspaceWithouthDirectives(workspaceBuffer, workspaceWithoutDirectives)
+	if err != nil {
+		return err
+	}
+
+	// write the directive-less workspace and update repos
+	if err := os.WriteFile("WORKSPACE", workspaceWithoutDirectives.Bytes(), 0666); err != nil {
+		return err
+	}
+
+	if verbose {
+		fmt.Println("Running update-repos outputting to temporary file.")
+	}
+	cmd := exec.CommandContext(ctx, "bazel", "run", "//:gazelle", "--", "update-repos", "-from_file=go.mod", fmt.Sprintf("-to_macro=%s%%gazelle_dependencies", _tmpBzl))
+	if out, err := cmd.CombinedOutput(); err != nil {
+		fmt.Println(string(out))
+		return err
+	}
+
+	depsMaybeBuff := new(bytes.Buffer)
+	workspaceDirectivesBuff := new(bytes.Buffer)
+	// parse the resulting tmp.bzl for deps.bzl and WORKSPACE updates
+	if verbose {
+		fmt.Println("Parsing temporary bzl file to prepare deps.bzl and WORKSPACE modifications.")
+	}
+	if err = getBuffsFromTmp(depsMaybeBuff, workspaceDirectivesBuff); err != nil {
+		return err
+	}
+
+	// update deps
+	depsFile, err := os.ReadFile("deps.bzl")
+	if err != nil {
+		return err
+	}
+	existingDepsScanner := bufio.NewScanner(bytes.NewBuffer(depsFile))
```

Comment:
```suggestion
	depsFile, err := os.Open("deps.bzl")
	if err != nil {
		return err
	}
	existingDepsScanner := bufio.NewScanner(depsFile)
```

If you are going to scan the file line by line, you can avoid reading the whole file into the memory